src/foam/dao/XMLDAO.java

```diff
@@ -0,0 +1,76 @@
+package foam.dao;
+
+import foam.core.FObject;
+import foam.core.XMLSupport;
+
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import javax.xml.stream.XMLStreamException;
+import foam.core.X;
+import foam.core.ProxyX;
+
+public class XMLDAO
+        extends MapDAO
+{
+
+    private String fileName;
+
+    public void setFileName(String filename) {
+        if (filename.contains(".xml")){
```

Comment:
fileName = System.getProperty("user.dir") + filename + (filename.contains(".xml") ? "" : ".xml");
Also, better to check if it endsWith() rather than contains. Safer and faster.